### PR TITLE
Improved grammar for kotlin based on official documentation

### LINKED ISSUE #3403: Improved support for textmate definition for kotlin dsl
I've made some improvements to the Kotlin grammar based on the official documentation and created a PR in the buildship repo.
It has been siting there for a month. Is there any interest in me porting the content of the PR? 

See: https://github.com/eclipse/buildship/pull/1273

### PR DESCRIPTION
**Pull request by per-steinar**
PR improves kotlin grammar based on documentation from [https://kotlinlang.org/docs/home.html](https://kotlinlang.org/docs/home.html)
Resolves #3403

### CODE REVIEW

**Comment by rgrunber** [comment]
As long as we've made an attempt to upstream the work, at https://github.com/eclipse/buildship/pull/1273, I'm fine with getting it in here sooner. @jdneo , feel free to merge if it all looks good to you.

**Comment by jdneo** [comment]
Hi @per-steinar, thank you for your contribution!

Looks like lots of cases will be improved with this PR. If possible, would you mind give the improved sample snippets as comment here?

**Review by jdneo — APPROVED** [review]
Tried on a simple gradle kotlin dsl: the token `mainClass` has a different color now

before:
before

after:
image

**Comment by jdneo** [comment]
Thank you @per-steinar!

File: language-support/kotlin/kotlin.tmLanguage.json [hunk 1]
```diff
@@ -21,10 +21,13 @@
     ],
     "repository": {
         "import": {
-            "begin": "\\b(import)\\b\\s*",
+            "begin": "\\b(import)\\b\\s?([\\w+.]*\\w+)?\\s*",
```
Comment (fwcd) [inline language-support/kotlin/kotlin.tmLanguage.json:24]: This rule unfortunately isn't quite correct and matches the package name as a modifier token, see: https://github.com/fwcd/vscode-kotlin/pull/137#discussion_r1451814478.

**Pull request merged**